Crash when specifying a cache with --cache

The OSG test suite sets up a cache on localhost:8001 and tries to download a file through it with the command
```
stashcp -d --cache=http://localhost:8001/ /osgtest/PUBLIC/testfile3 /tmp/tmpiord6aip
```
on 6.8.0 this fails with errors and a stack trace
```
time="2022-08-09T10:53:43-05:00" level=debug msg="Len of source: 2"
time="2022-08-09T10:53:43-05:00" level=debug msg="Sources: [/osgtest/PUBLIC/testfile3]"
time="2022-08-09T10:53:43-05:00" level=debug msg="Destination: /tmp/tmpiord6aip"
time="2022-08-09T10:53:43-05:00" level=error msg="Failed to download namespaces: 404 NOT FOUND"
time="2022-08-09T10:53:43-05:00" level=debug msg="Failed to download namespaces: Failed to download namespaces: 404 NOT FOUND, continueing using built-in namespace configuration"
<dump of namespaces file>
time="2022-08-09T10:53:43-05:00" level=info msg="Trying CVMFS..."
time="2022-08-09T10:53:43-05:00" level=debug msg="Checking if the CVMFS file exists: /cvmfs/stash.osgstorage.org/osgtest/PUBLIC/testfile3"
time="2022-08-09T10:53:43-05:00" level=debug msg="CVMFS File does not exist"
time="2022-08-09T10:53:43-05:00" level=info msg="Trying XROOTD..."
time="2022-08-09T10:53:43-05:00" level=info msg="Trying HTTP..."
time="2022-08-09T10:53:43-05:00" level=debug msg="Nearest cache list: [http://localhost:8001/]"
time="2022-08-09T10:53:43-05:00" level=debug msg="Cache list name: []"
time="2022-08-09T10:53:43-05:00" level=debug msg="Matched caches: []"
time="2022-08-09T10:53:43-05:00" level=debug msg="Trying the caches: []"
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/opensciencegrid/stashcp/v6.download_http({0x7ffda24e3f32, 0x0}, {0x7ffda24e3f4c, 0x10}, 0x0, {{0x0, 0x0, 0x0}, {0x0, 0x0}, ...}, ...)
	/home/runner/work/stashcp/stashcp/handle_http.go:214 +0xa8c
github.com/opensciencegrid/stashcp/v6.DoStashCPSingle({0x7ffda24e3f32, 0x19}, {0x7ffda24e3f4c, 0x10}, {0xc00005b560, 0x3, 0x0}, 0x0)
	/home/runner/work/stashcp/stashcp/main.go:293 +0xd49
main.main()
	/home/runner/work/stashcp/stashcp/cmd/stashcp/main.go:222 +0xf11
```
perhaps because the cache is not listed in the namespaces JSON?

I think if a cache is explicitly specified then it should always be used instead of doing any sort of lookup.  (Also don't try CVMFS if it's an HTTP URL.)

--- a/cmd/stashcp/main.go
+++ b/cmd/stashcp/main.go
@@ -209,6 +209,11 @@ func main() {
 	// Convert the methods
 	splitMethods := strings.Split(options.Methods, ",")
 
+	// If the user overrides the cache, then only use HTTP
+	if stashcp.CacheOverride {
+		splitMethods = []string{"http"}
+	}
+
 	if len(source) > 1 {
 		if destStat, err := os.Stat(dest); err != nil && destStat.IsDir() {
 			log.Errorln("Destination is not a directory")

--- a/handle_http.go
+++ b/handle_http.go
@@ -182,7 +182,7 @@ func download_http(source string, destination string, payload *payloadStruct, na
 
 	// Now that we have the ordered list of caches, do an intersect for the caches for the namespace
 	var closestNamespaceCaches []Cache
-	if CacheOverride == true {
+	if CacheOverride {
 		cache := Cache{
 			Endpoint:     NearestCache,
 			AuthEndpoint: NearestCache,

--- a/handle_http.go
+++ b/handle_http.go
@@ -181,7 +181,17 @@ func download_http(source string, destination string, payload *payloadStruct, na
 	log.Debugln("Cache list name:", namespace.Caches)
 
 	// Now that we have the ordered list of caches, do an intersect for the caches for the namespace
-	closestNamespaceCaches := namespace.MatchCaches(NearestCacheList)
+	var closestNamespaceCaches []Cache
+	if CacheOverride == true {
+		cache := Cache{
+			Endpoint:     NearestCache,
+			AuthEndpoint: NearestCache,
+			Resource:     NearestCache,
+		}
+		closestNamespaceCaches = []Cache{cache}
+	} else {
+		closestNamespaceCaches = namespace.MatchCaches(NearestCacheList)
+	}
 	log.Debugln("Matched caches:", closestNamespaceCaches)
 
 	// Make sure we only try as many caches as we have

